Watchman.Discord/WatchmanBot.cs

```diff
@@ -39,25 +40,23 @@ public WatchmanBot(DiscordConfiguration configuration)
                 TotalShards = 1
             });
             this._client.MessageReceived += this.MessageReceived;
-
+            
             this._container = GetAutofacContainer(configuration);
             Log.Logger = SerilogInitializer.Initialize(this._container.Resolve<IMongoDatabase>(), this.LogOnChannel);
             Log.Information("Bot created...");
-            this._workflow = GetWorkflow(configuration, _container);
+            this._workflow = GetWorkflow(configuration);
         }
 
         public async Task Start()
         {
             MongoConfiguration.Initialize();
-            ServerInitializer.Initialize(_client, _container.Resolve<MessagesServiceFactory>());
+            ServerInitializer.Initialize(_client);
 
             _ = Task.Run(DefaultHelpInit);
-            
+            AssignEvents();
+
             await _client.LoginAsync(TokenType.Bot, this._configuration.Token);
```

Comment:
troche to wykracza poza tego taska, ale

myśle że ok opcją byłby workflow builder 
który z ładną składnią konfigurowałby wszystko dotyczące klienta discord.net i naszego workflow
robienie await _client.LoginAsync(TokenType.Bot, this._configuration.Token); 
albo await Task.Delay(-1);

wygląda jak krzywe akcje 

powinien być builder który konfiguruje Workflow i klienta na podstawie konfiguracji ustawianej w metodach 
na koniec ma metode Build() 
która zwraca nam wspomniane wcześniej await Task.Delay(-1);

przykładowo byłoby to coś typu 

return WorkflowBuilder.Create(token)
.MetodaUstawiajacaMiddleware(options => 
    options.AddMiddleware<jakieśMiddleware>();
    options.AddMiddleware<jakieśInneMiddleware>();)
.MetodaUstawiającaCośNpHandleryOnReadItp(taki zapis jak przy middleware)
może inne rzeczy jeśli potrzebujemy
.Build()

to tylko prosty schemat jak ja to widze
zamiast zapisu z options równie dobrze można odpalać wszystko osobno

główny cel to wygoda 
osoba używająca frameworka nie chce sie przejmować jakimś klientem pod spodem, chce podać token i mieć spokój 
a do tego my zyskujemy na tym że mamy porządek w bocie i klasa WatchmanBot jest przyjazna 

dodatkowo ok opcją byłoby dodanie czegoś jak AddDefaultMiddlewares, żeby te middleware podstawowe ustawiały sie zawsze 
a dodatkowe powinny być ustawiane tylko jak użytkownik chce dorzucić coś własnego i niestandardowego